SIL.Windows.Forms.Keyboarding/Linux/UnityXkbKeyboardSwitchingAdaptor.cs

```diff
@@ -16,11 +16,13 @@ public UnityXkbKeyboardSwitchingAdaptor(IXklEngine engine): base(engine)
 		protected override void SelectKeyboard(KeyboardDescription keyboard)
 		{
 			var xkbKeyboard = keyboard as XkbKeyboardDescription;
-			if (xkbKeyboard != null)
-			{
-				if (xkbKeyboard.GroupIndex >= 0)
-					UnityIbusKeyboardSwitchingAdaptor.SelectKeyboard((uint)xkbKeyboard.GroupIndex);
-			}
+			if (xkbKeyboard == null || xkbKeyboard.GroupIndex < 0)
+				return;
+
+			var switchingAdaptor = KeyboardController.Instance
```

Comment:
I'm actually a bit confused about how this is supposed to be working, and am having trouble figuring it out from just the github UI. I need to look at it in an IDE.

Reply:
So how do you know that `SwitchingAdaptor` will be an `IUnityKeyboardSwitchingAdaptor` and not result in a null reference exception on the next line?

I think I'm confused about why the `UnityXkbKeyboardSwitchingAdaptor` would be being used anyway if keyboard switching is being done thru ibus.